subprojects/plugins/src/test/groovy/org/gradle/api/internal/plugins/UnixStartScriptGeneratorTest.groovy

```diff
@@ -50,6 +51,20 @@ class UnixStartScriptGeneratorTest extends Specification {
         destination.toString().contains(TextUtil.unixLineSeparator)
     }
 
+    @Issue("https://github.com/gradle/gradle/issues/25036")
+    def "unix script ignores output from cd command"() {
```

Comment:
I feel like this test is too heavily tied to the implementation of the script. It would be better to actually run a generated copy of the script and see if it produces unwanted output, i.e. test the expected behavior, not the solution.

Reply:
FYI, this is a regression coming from [this commit](https://github.com/kurahaupo/gradle/commit/e3c8398bc51b48b988cb34846984720db13bddd9#diff-a549a2629a3d96f76e3f043a1526e48fb086e90e9d9d03ae0f7abf6d3e803333L37-R43). You can see the redirection to `/dev/null` was present before the refactor.

Given the complexity of testing this very special case, would it be acceptable to submit a new PR without a test and simply add a comment in the code?

Reply:
I think making those changes in this PR should be fine.